distributed/cli/dask_submit.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import os
+from time import sleep
+
+import click
+from tornado import gen
+from tornado.ioloop import IOLoop
+
+from distributed import rpc, sync
+from distributed.cli.utils import check_python_3
+
+logger = logging.getLogger(__name__)
+
+
+class Submit(object):
+    def __init__(self, ip, port, loop=None):
+        self.loop = loop or IOLoop()
+        self.ip = ip
+        self.port = port
+        self.coroutines = []
+        self.start()
+
+    def start(self, **kwargs):
+        sync(self.loop, self._start, **kwargs)
+
+    def __enter__(self):
+        if not self.loop._running:
+            self.start()
+        return self
+
+    @gen.coroutine
+    def _start(self):
+        self.remote_client = rpc(ip=self.ip, port=self.port)
+
+    def run(self, filepath=None):
+        remote_file = os.path.basename(filepath)
+        self.remote_client.upload_file(filename=remote_file, source_file_path=filepath)
+        return self.remote_client.execute(filename=remote_file)
```

Comment:
I believe that this could all be collapsed down to a single coroutine.  Perhaps something like the following:

```python
@gen.coroutine
def submit(ip, port, filepath):
    rc = rpc(ip=ip, port=port)
    yield rc.upload_file(filename=remote_file, source_file_path=filepath)
    result = rc.execute(filename=remote_file)
    raise gen.Return(result)
```